lib/src/view/tournament/tournament_screen.dart

```diff
@@ -0,0 +1,489 @@
+          ),
+      ],
+    );
+  }
+}
+
+class _Verdicts extends ConsumerWidget {
+  const _Verdicts(this.verdicts);
+
+  final Verdicts verdicts;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final isLoggedIn = ref.watch(authSessionProvider)?.user.id != null;
+
+    if (verdicts.list.isEmpty) {
+      return const SizedBox.shrink();
+    }
+
+    return Row(
+      children: [
+        Icon(
+          isLoggedIn && verdicts.accepted ? Icons.check : Icons.lock,
+          color:
+              isLoggedIn
+                  ? verdicts.accepted
+                      ? LichessColors.good
+                      : LichessColors.error
+                  : null,
+          size: 30,
+        ),
+        const SizedBox(width: 10),
+        Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: verdicts.list
+              .whereNot((verdict) => verdict.condition == 'Bot players are not allowed')
+              .map(
+                (verdict) => Text(
+                  verdict.condition,
+                  style: TextStyle(
+                    color:
+                        isLoggedIn
+                            ? verdict.ok
+                                ? LichessColors.good
+                                : LichessColors.error
+                            : null,
+                  ),
+                ),
+              )
+              .toList(growable: false),
+        ),
+      ],
+    );
+  }
+}
+
+class _FeaturedGame extends ConsumerWidget {
+  const _FeaturedGame(this.featuredGame);
+
+  final FeaturedGame featuredGame;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return LayoutBuilder(
+      builder: (context, constraints) {
+        final boardSize = constraints.maxWidth;
+        switch (ref.watch(
+          tvControllerProvider(null, (featuredGame.id, featuredGame.orientation)),
+        )) {
+          case AsyncData(:final value):
+            {
+              final game = value.game;
+
+              final whitePlayer = _FeaturedGamePlayer(
+                game: game,
+                player: featuredGame.white,
+                side: Side.white,
+              );
+
+              final blackPlayer = _FeaturedGamePlayer(
+                game: game,
+                player: featuredGame.black,
+                side: Side.black,
+              );
+
+              return GestureDetector(
+                child: BoardThumbnail(
+                  size: boardSize,
+                  orientation: featuredGame.orientation,
+                  fen: game.lastPosition.fen,
+                  header: featuredGame.orientation == Side.white ? blackPlayer : whitePlayer,
+                  footer: featuredGame.orientation == Side.white ? whitePlayer : blackPlayer,
+                  lastMove: game.lastMove,
+                ),
+              );
+            }
+          case _:
+            return BoardThumbnail(
```

Comment:
It is more efficient to always return the GestureDetector in the loading state, so only the onTap handler changes.

Reply:
Huh, I just noticed the `GestureDetector` doesn't actually have a `onTap` handler here.  I probably wanted it to open the game screen, but do we really want that? Players might miss their next pairing (as the websocket connection gets closed once you leave the tournament screen) if they open the feature game and enter the game screen, so let's just remove the `GestureDetector` here

Reply:
No I meant when flutter does the tree diff to see what's changed, it is more efficient to always keep the `GestureDetector`, but in the loading state the onTap is null or does nothing.

If you change the tree, first no GestureDetector, then GestureDetector, flutter has more work to do.

That is nitpicking, but it is a good habit to take (even though on some cases we cannot avoid it).

Reply:
Ah, that makes sense, will keep in mind!

Still, the GestureDetector was actually redundant here, so I removed it entirely